packages/fxa-auth-server/lib/routes/subscriptions/mozilla.ts

```diff
@@ -0,0 +1,108 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { ServerRoute } from '@hapi/hapi';
+import { Container } from 'typedi';
+import error from '../../error';
+import { PaymentBillingDetails, StripeHelper } from '../../payments/stripe';
+import { CapabilityService } from '../../../lib/payments/capability';
+import { MozillaSubscription } from '../../../../fxa-shared/subscriptions/types';
+import { AuthLogger, AuthRequest } from '../../types';
+import {
+  handleAuth,
+  subscribedProductIdsToIapGooglePlaySubscriptions,
+} from './utils';
+import validators from '../validators';
+
+export const mozillaSubscriptionRoutes = ({
+  log,
+  db,
+  customs,
+  stripeHelper,
+  capabilityService,
+}: {
+  log: AuthLogger;
+  db: any;
+  customs: any;
+  stripeHelper: StripeHelper;
+  capabilityService?: CapabilityService;
+}): ServerRoute[] => {
+  if (!capabilityService) {
+    capabilityService = Container.get(CapabilityService);
+  }
+  const mozillaSubscriptionHandler = new MozillaSubscriptionHandler(
+    log,
+    db,
+    customs,
+    stripeHelper,
+    capabilityService
+  );
+  return [
+    {
+      method: 'GET',
+      path: '/oauth/mozilla-subscriptions/customer',
+      options: {
+        auth: {
+          payload: false,
+          strategy: 'oauthToken',
+        },
+        response: {
+          schema: validators.subscriptionsMozillaSubscriptionsValidator,
+        },
+      },
+      handler: (request: AuthRequest) =>
+        mozillaSubscriptionHandler.getCustomer(request),
+    },
+  ];
+};
+
+export class MozillaSubscriptionHandler {
+  constructor(
+    protected log: AuthLogger,
+    protected db: any,
+    protected customs: any,
+    protected stripeHelper: StripeHelper,
+    protected capabilityService: CapabilityService
+  ) {}
+
+  /**
+   * This misnomer is consistent with the original from StripeHandler.  The
+   * function returns an object with (optional) Stripe billing details and a
+   * list of subscriptions, including IAP ones.
+   */
+  async getCustomer(request: AuthRequest) {
```

Comment:
Would it be really painful to rename this method and the one in the StripeHandler? Maybe worth filing a follow-up?